docs/essentials/gc/unmanaged/implementingdispose.md

```diff
@@ -1 +1,442 @@
+
+Aggressive garbage collection can cause a finalizer to run while a member of the reclaimed object is still executing. It is a good idea to call the [GC.KeepAlive](http://dotnet.github.io/api/System.GC.html#System_GC_KeepAlive_System_Object_) method at the end of a lengthy Dispose method.
+
+## Dispose() and Dispose(Boolean)
+
+The [IDisposable](http://dotnet.github.io/api/System.IDisposable.html) interface requires the implementation of a single parameterless method, [Dispose()](http://dotnet.github.io/api/System.IDisposable.html#System_IDisposable_Dispose). However, the dispose pattern requires two `Dispose` methods to be implemented: 
+
+* A public non-virtual `IDisposable.Dispose` implementation that has no parameters.
+
+* A protected virtual `Dispose` method whose signature is:
+
+```csharp
+protected virtual void Dispose(bool disposing)
+```
+
+### The Dispose() overload
+
+Because the public, non-virtual, parameterless [Dispose()](http://dotnet.github.io/api/System.IDisposable.html#System_IDisposable_Dispose) method is called by a consumer of the type, its purpose is to free unmanaged resources and to indicate that the finalizer, if one is present, doesn't have to run. Because of this, it has a standard implementation:
+
+```csharp
+public void Dispose()
+{
+   // Dispose of unmanaged resources.
+   Dispose(true);
+   // Suppress finalization.
+   GC.SuppressFinalize(this);
+}
+```
+
+The 'Dispose' method performs all object cleanup, so the garbage collector no longer needs to call the objects' [Object.Finalize](http://dotnet.github.io/api/System.Object.html#System_Object_Finalize) override. Therefore, the call to the [SuppressFinalize](http://dotnet.github.io/api/System.GC.html#System_GC_SuppressFinalize_System_Object_) method prevents the garbage collector from running the finalizer. If the type has no finalizer, the call to `GC.SuppressFinalize` has no effect. Note that the actual work of releasing unmanaged resources is performed by the second overload of the `Dispose` method
+
+### The Dispose(Boolean) overload
+
+In the second overload, the *disposing* parameter is a `Boolean` that indicates whether the method call comes from a `Dispose` method (its value is `true`) or from a finalizer (its value is `false`). 
+
+The body of the method consists of two blocks of code: 
+
+* A block that frees unmanaged resources. This block executes regardless of the value of the *disposing* parameter. 
+
+* A conditional block that frees managed resources. This block executes if the value of *disposing* is `true`. The managed resources that it frees can include: 
+
+    * Managed objects that implement `IDisposable`.  The conditional block can be used to call their `Dispose` implementation. If you have used a safe handle to wrap your unmanaged resource, you should call the `SafeHandle.Dispose(Boolean)` implementation here. 
```

Comment:
An object that owns some `SafeHandle` can't call `SafeHandle.Dispose(Boolean)`, because it's `protected`.